contracts/script/DeployFeeContract.s.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "forge-std/Script.sol";
+
+import { FeeContract } from "../src/FeeContract.sol";
+import { ERC1967Proxy } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+
+contract DeployFeeContract is Script {
+    function run() external returns (address payable proxy, address admin) {
+        // string memory seedPhrase = vm.envString("MNEMONIC");
+        // (address admin,) = deriveRememberKey(seedPhrase, 0);
+        (proxy, admin) = deployFeeContract();
+
+        return (proxy, admin);
+    }
+
+    /// @notice deploys the impl, proxy & initializes the impl
+    // @return address of proxy and admin
+    function deployFeeContract() public returns (address payable proxyAddress, address admin) {
+        vm.startBroadcast();
+
+        FeeContract feeContract = new FeeContract(); //Our implementation(logic).Proxy will point
+            // here to delegate
+
+        // Encode the initializer function call
+        bytes memory data = abi.encodeWithSelector(
+            FeeContract(feeContract).initialize.selector,
+            msg.sender // Initial owner/admin of the contract
+        );
+
```

Comment:
I actually didn't fully understand how this works. 
[`ERC1967Proxy.constructor`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/ERC1967/ERC1967Proxy.sol#L27) will call `ERC1967Utils.upgradeToAndCall` which [would call ](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/ERC1967/ERC1967Utils.sol#L86)`Address.functionDelegateCall`, which [would call ](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Address.sol#L105)the constructor on fee contract with `msg.sender`

but our fee contract constructor doesn't accept any input argument

what did I miss?

Reply:
Yea the FeeContract's constructor is actually disabled, so the function that's actually being called is the `initialize` method on `FeeContract.sol`

Reply:
oh yeaaa, good point, it just uses the msg.sender that calls the method, so I can comment that code out

Reply:
https://github.com/EspressoSystems/espresso-sequencer/commit/7df709c6ea1e027957ebd43e507e080eb4a7e2d0